src/firefly/js/tables/TableRequestUtil.js

```diff
@@ -160,6 +165,45 @@ export function makeVOCatalogRequest(title, params={}, options={}) {
 
 const voProviders = {'NED':'NedSearch'};
 
+/**
+ * Parameters for cone search
+ * @typedef {object} ResourceInfo
+ * @global
+ * @public
+ * @prop {TableRequest} request     required. the table request to create this resource from
+ * @prop {string}       [scope]       access type of this resource.  one of 'global', 'user', 'protected'.  defaults to 'global'.
+ * @prop {string}       [secret]      the secret token required to access 'protected' scope
+ */
+
+/**
+ * Creates a table request object for the given resource
+ * @param {object} p                function parameters
+ * @param {ResourceInfo} p.resource resource information
+ * @param {string} [p.title]        title to display with this table.
+ * @param {TableRequest} [p.options] table request options.  see TableRequest for details.
+ * @param {string} p.action  one of 'query', 'create', 'delete'.  defautls to 'query'.
+ *                         when action is 'create' or 'delete', it will submit the request.
+ *                         no further action is needed.
+ * @returns {TableRequest}
+ * @public
+ * @func  makeResourceRequest
+ * @memberof firefly.util.table
+ */
+export function makeResourceRequest({resource={}, title, options={}, action='query'}) {
```

Comment:
I think it function is too overloaded. It makes it API confusing. I think you should break it up it 

1. `makeResourceRequest` for query, which is consistent with how we do things now.
2. `createResource`
3. `deleteResource`

if you could come up with a good name then `createResource` and `deleteResource` could be combined (`putResource`?)

I had to really study your example to understand what was happning.

Reply:
That's what I had before.  `makeResourceRequest `,  and `createResource`.  
Then, I added the `delete` feature.  In order to not expose the `action` param, I need to add a third function, `deleteResource`.  That's when I decided to combine hem all and expose the `action` param.

In the simplest case, you can do this.  (original request)
```
const aResource = {request: makeFileRequest(null, 'https://web.ipac.caltech.edu/staff/roby/demo/WiseDemoTable.tbl')};
const myTbl =   makeResourceRequest({resource: aResource, title: 'My Table'});
firefly.showTable('actual', aResource);
```
I have no problem separate them into 3 functions.  After seeing my explanation, do you still think I should separate them?

Reply:
I talk to @loitly, he will break it into three calls, after that merge.